client/src/webview-js/image-upload/image-upload.ts

```diff
@@ -1,22 +1,30 @@
-let vscode
-let pending = []
+declare let acquireVsCodeApi: any
+let vscode: any
+
+interface PendingEntry {
+  mediaName: string
+  data: string | ArrayBuffer | null
+}
+let pending: PendingEntry[] = []
 
 window.addEventListener('load', () => {
   vscode = acquireVsCodeApi() // eslint-disable-line no-undef
 
-  const dropArea = document.getElementById('drop-area')
-  const preview = document.getElementById('preview')
-  const uploadButton = document.getElementById('trigger-upload')
-  const cancelButton = document.getElementById('cancel-upload')
+  const dropArea = document.getElementById('drop-area') as HTMLElement
+  const preview = document.getElementById('preview') as HTMLElement
+  const uploadButton = document.getElementById('trigger-upload') as HTMLElement
+  const cancelButton = document.getElementById('cancel-upload') as HTMLElement
 
-  const clearPending = () => {
+  const clearPending = (): void => {
     pending = []
-    while (preview.firstChild) {
-      preview.firstChild.remove()
+    if (preview.firstChild !== null) {
+      while (preview.firstChild) { // eslint-disable-line @typescript-eslint/strict-boolean-expressions
```

Comment:
I think the linter probably wants something like:
```suggestion
      while (preview.firstChild != null) {
```
which would also allow you to drop the `if`